src/main/java/games/strategy/engine/framework/startup/login/ClientLogin.java

```diff
@@ -33,6 +45,20 @@ public ClientLogin(final Component parent) {
           "Enter a password to join the game", JOptionPane.QUESTION_MESSAGE);
       final String password = new String(passwordField.getPassword());
       rVal.put(PASSWORD_PROPERTY, MD5Crypt.crypt(password, challengProperties.get(ClientLoginValidator.SALT_PROPERTY)));
+      final String publicKeyString = challengProperties.get(ClientLoginValidator.RANDOM_RSA_PUBLIC_KEY_PROPERTY);
+      if (publicKeyString != null) {
+        try {
+          final PublicKey publicKey = KeyFactory.getInstance(ClientLoginValidator.RSA)
+              .generatePublic(new X509EncodedKeySpec(Base64.getDecoder().decode(publicKeyString)));
+          final Cipher cipher = Cipher.getInstance(ClientLoginValidator.RSA_ECB_OAEPP);
+          cipher.init(Cipher.ENCRYPT_MODE, publicKey);
+          rVal.put(ENCRYPTED_PASSWORD_PROPERTY,
+              Base64.getEncoder().encodeToString(cipher.doFinal(password.getBytes())));
+        } catch (InvalidKeySpecException | NoSuchAlgorithmException | IllegalBlockSizeException | BadPaddingException
+            | InvalidKeyException | NoSuchPaddingException e) {
```

Comment:
Could we just handle `GeneralSecurityException` here since we're treating all these exception types the same?